lib/pdk/cli/util/changelog_generator.rb

```diff
@@ -0,0 +1,58 @@
+module PDK
+  module CLI
+    module Util
+      class ChangelogGenerator
+        def generate_changelog
+          retries = 0
+          result = nil
+          loop do
+            result = system('bundle exec rake changelog')
+            retries += 1
+            break if result || retries > 3
+          end
+
+          unless result
+            raise 'Error generating changelog'
+          end
+
+          output = File.read('CHANGELOG.md')
+
+          raise 'Uncategorized PRs; Please go label them' if output =~ %r{UNCATEGORIZED PRS; GO LABEL THEM}
+        end
+
+        def get_next_version(current_version)
+          current_version_map = current_version.split('.').map { |v| v.to_i }
+          puts '# Extracting target version from CHANGELOG.md'
+
+          # Grab all lines that start with ## between two lines that start with the version to detect changes
+          data = ''
+          matches = 1
+          File.foreach('CHANGELOG.md') do |line|
```

Comment:
Assumes `pwd` is in a module.  This is a dangerous assumption.

Reply:
This utility class is used in the release command, where there is an ensure that the pwd is in the module, see line: https://github.com/puppetlabs/pdk/pull/750/files#diff-87ad85e8da7245ccfda95cfed0b88327R19

Reply:
This still feels like it needs work.  What if the file doesn't exist? What if the file is empty? What if the file isn't encoded correctly? What encoding does it even use? Will it break lines correctly on Windows (CRLF) and Linux (LF) correctly?

Reply:
Using PDK::Util::Filesystem.read_file now